src/postgres/base-pg-store.ts

```diff
@@ -73,4 +74,40 @@ export class BasePgStore {
   ): Promise<UnwrapPromiseArray<T>> {
     return this.sqlTransaction(callback, false);
   }
+
+  /**
+   * Refreshes a materialized view concurrently depending on the current environment.
+   * @param viewName - Materialized view name
+   */
+  async refreshMaterializedView(viewName: string) {
+    await this.sql`REFRESH MATERIALIZED VIEW ${
+      isProdEnv ? this.sql`CONCURRENTLY` : this.sql``
+    } ${this.sql(viewName)}`;
+  }
+}
+
+/**
+ * Base module that extends PgStore functionality and allows organizing queries in separate files.
+ */
+export class BasePgStoreModule {
+  private readonly parent: BasePgStore;
+
+  constructor(db: BasePgStore) {
+    this.parent = db;
+  }
+
+  protected get sql(): PgSqlClient {
+    return this.parent.sql;
+  }
+  protected async sqlTransaction<T>(
+    callback: (sql: PgSqlClient) => T | Promise<T>,
+    readOnly = true
+  ): Promise<UnwrapPromiseArray<T>> {
+    return this.parent.sqlTransaction(callback, readOnly);
+  }
+  async sqlWriteTransaction<T>(
```

Comment:
Is there a reason for not using `protected` here but on readonly?

Reply:
Yep, sorry, it was wrong copy pasta